examples/ingestion-patterns/README.md

```diff
@@ -0,0 +1,250 @@
+# Data Ingestion Patterns - Dagster Example Project
+
+A Dagster project demonstrating three fundamental data ingestion patterns: **Push**, **Pull**, and **Poll**.
+
+## Project Structure
+
+```
+ingestion-patterns/
+├── src/ingestion_patterns/           # Main Python package
+│   ├── definitions.py                # Autoloading definitions with Definitions.merge
+│   ├── defs/                         # Assets, jobs, sensors, schedules
+│   │   ├── jobs.py                   # Jobs, schedules, and sensors
+│   │   ├── push_webhook_ingestion.py # Push pattern (webhooks)
+│   │   ├── pull_api_ingestion.py     # Pull pattern (scheduled API)
+│   │   └── poll_kafka_ingestion.py   # Poll pattern (Kafka)
+│   └── resources/
+│       └── mock_apis.py              # Mock API clients and webhook storage
+├── scripts/
+│   ├── kafka_producer.py             # Produce events to Kafka
+│   ├── webhook_server.py             # Flask webhook receiver server
+│   └── webhook_simulator.py          # Simulate webhooks (in-memory)
+├── tests/
+│   ├── test_definitions.py           # Definition loading tests
+│   └── test_assets.py                # Asset materialization tests
+├── docker-compose.yml                # Kafka + Kafka UI
+└── pyproject.toml                    # Project configuration
+```
+
+## Quick Start
+
+### 1. Install Dependencies
+
+```bash
+cd examples/ingestion-patterns
+
+# Install with uv (recommended)
+uv sync --all-groups
+
+# Or install manually
+uv pip install -e .
+```
+
+### 2. Start Dagster
+
+```bash
+dg dev
+```
+
+Open http://localhost:3000 to view the Dagster UI.
+
+### 3. Materialize Assets
+
+In the Dagster UI:
+
+1. Navigate to the **Assets** page
+2. Select assets to materialize
+3. Click **Materialize** to run
+
+## Ingestion Patterns
+
+### Push Pattern (Webhooks)
+
+External systems push data to your platform via webhooks.
+
+**Asset:** `process_webhook_data`
+**Sensor:** `webhook_pending_sensor` (checks for pending payloads every 10s)
+
+#### Demo: Simulate Webhooks
+
+**Option A: Use the simulator script (in-memory)**
+
+```bash
+python scripts/webhook_simulator.py --count 10
+```
+
+Then materialize `process_webhook_data` in the Dagster UI.
+
+**Option B: Run the webhook server**
+
+```bash
+# Terminal 1: Start the webhook server
```

Comment:
It might be cleaner if the webhook server was just another service in the docker compose